lib/plugins/mk{Neovim,Vim}Plugin: add dependencies parameter (#3231)
nix-community/nixvim: CONTRIBUTING.md

| **callSetup**                | Indicating whether to call the setup function. Useful when `setup` function needs customizations.                                                                                                                                                    | No       | `true`                                                                                          |
| **colorscheme**              | The name of the colorscheme.                                                                                                                                                                                                                         | No       | `name` parameter                                                                                |
| **configLocation**           | The option location where the lua configuration should be installed. Nested option locations can be represented as a list. The location can also be wrapped using `lib.mkBefore`, `lib.mkAfter`, or `lib.mkOrder`.                                   | No       | `"extraConfigLuaPre"` if `isColorscheme` then `extraConfigLuaPre`, otherwise `"extraConfigLua"` |

> | **dependencies**             | Which dependencies to enable by default with this plugin. (List of strings)                                                                                                                                                                          | No       | `"extraConfigLuaPre"` if `isColorscheme` then `extraConfigLuaPre`, otherwise `"extraConfigLua"` |

Reviewer: ```suggestion
| **dependencies**             | A list of dependencies to enable by default with this plugin. (List of strings)                                                                                                                                                                          | No       | `[ ]` |
```

(whitespace is probably wrong)

Should we link `dependencies` to the option?

```nix
| **dependencies**             | A list of [`dependencies`] to enable by default with this plugin. (List of strings)                                                                                                                                                                          | No       | `[ ]` |

[`dependencies`]: https://nix-community.github.io/nixvim/dependencies/bat.html
```

| **name**                     | The name of the Vim plugin.                                                                                                                                                                                                                        | Yes      | N/A                                   |
| **url**                      | The URL of the plugin repository.                                                                                                                                                                                                                  | Yes      | `package` parameter's `meta.homepage` |
| **colorscheme**              | The name of the colorscheme.                                                                                                                                                                                                                       | No       | `name` parameter                      |

> | **dependencies**             | Which dependencies to enable by default with this plugin. (List of strings)                                                                                                                                                                          | No       | `"extraConfigLuaPre"` if `isColorscheme` then `extraConfigLuaPre`, otherwise `"extraConfigLua"` |

Reviewer: ```suggestion
| **dependencies**             | A list of dependencies to enable by default with this plugin. (List of strings)                                                                                                                                                                          | No       | `[ ]` |
```

(whitespace is probably wrong)

(Again, should we have a `` [`dependencies`] `` link?)
